src/components/Custom/RJSFFormFieldAdapter/CustomPriorityBoundsField.jsx

```diff
@@ -169,12 +172,140 @@ const CustomPriorityBoundsField = (props) => {
     }
   }, [isMapVisible]);
 
+  // Close tooltip when clicking outside
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (showTooltip && !event.target.closest(".tooltip-container")) {
+        setShowTooltip(false);
+      }
+    };
+
+    document.addEventListener("click", handleClickOutside);
+    return () => document.removeEventListener("click", handleClickOutside);
+  }, [showTooltip]);
+
   const handleChange = (newData) => {
     if (typeof props.onChange === "function") {
       props.onChange(Array.isArray(newData) ? [...newData] : newData);
     }
   };
 
+  // Validate GeoJSON structure
+  const validateGeoJSON = (geoJson) => {
+    if (!geoJson || typeof geoJson !== "object") {
+      return { valid: false, error: "Invalid JSON format" };
+    }
+
+    if (!Array.isArray(geoJson.features)) {
+      return { valid: false, error: "Features must be an array" };
+    }
+
+    const polygonFeatures = geoJson.features.filter(
+      (feature) =>
+        feature.type === "Feature" && feature.geometry && feature.geometry.type === "Polygon",
+    );
+
+    if (polygonFeatures.length === 0) {
+      return { valid: false, error: "No valid Polygon features found" };
+    }
+
+    return { valid: true, features: polygonFeatures };
+  };
+
+  // Handle file upload
+  const handleFileUpload = async (file) => {
+    setUploadFeedback(null);
+
+    // Check file type
+    const validExtensions = [".json", ".geojson"];
+    const fileExtension = file.name.toLowerCase().substring(file.name.lastIndexOf("."));
+
+    if (!validExtensions.includes(fileExtension)) {
+      setUploadFeedback({ type: "error", message: messages.fileTypeError });
+      return;
+    }
+
+    try {
+      const text = await file.text();
+      const geoJson = JSON.parse(text);
+
+      const validation = validateGeoJSON(geoJson);
+      if (!validation.valid) {
+        setUploadFeedback({
+          type: "error",
+          message: messages.invalidGeoJSON,
+          details: validation.error,
+        });
+        return;
+      }
+
+      // Convert GeoJSON features to the format expected by the component
+      const convertedFeatures = validation.features.map((feature) => ({
+        type: "Feature",
+        geometry: feature.geometry,
+        properties: feature.properties || {},
+      }));
```

Comment:
You should remove this transformation, because if the input object was valid GeoJSON to begin with, then it does nothing.